Source/Core/DolphinQt2/Config/PathDialog.cpp

```diff
@@ -0,0 +1,190 @@
+			tr("Select Wii NAND Root"),
+			QDir::currentPath());
+	if (!dir.isEmpty())
+	{
+		m_nand_edit->setText(dir);
+		Settings().SetWiiNAND(dir);
+	}
+}
+
+QGroupBox* PathDialog::MakeGameFolderBox()
+{
+	QGroupBox* game_box = new QGroupBox(tr("Game Folders"));
+	game_box->setMinimumSize(QSize(400, 250));
+	QVBoxLayout* vlayout = new QVBoxLayout;
+
+	m_path_list = new QListWidget;
+	m_path_list->insertItems(0, Settings().GetPaths());
+	m_path_list->setSpacing(1);
+	vlayout->addWidget(m_path_list);
+
+	QHBoxLayout* hlayout = new QHBoxLayout;
+
+	hlayout->addStretch();
+	QPushButton* add = new QPushButton(tr("Add"));
+	QPushButton* remove = new QPushButton(tr("Remove"));
+	hlayout->addWidget(add);
+	hlayout->addWidget(remove);
+	vlayout->addLayout(hlayout);
+
+	connect(add, &QPushButton::clicked, this, &PathDialog::Browse);
+	connect(remove, &QPushButton::clicked, this, &PathDialog::RemovePath);
+
+	game_box->setLayout(vlayout);
+	return game_box;
+}
+
+QGridLayout* PathDialog::MakePathsLayout()
+{
+	QGridLayout* layout = new QGridLayout;
+	layout->setColumnStretch(1, 1);
+
+	m_game_edit = new QLineEdit(Settings().GetDefaultGame());
+	connect(m_game_edit, &QLineEdit::editingFinished,
+			[=]{ Settings().SetDefaultGame(m_game_edit->text()); });
+	QPushButton* game_open = new QPushButton;
+	connect(game_open, &QPushButton::clicked, this, &PathDialog::BrowseDefaultGame);
+	layout->addWidget(new QLabel(tr("Default Game")), 0, 0);
+	layout->addWidget(m_game_edit, 0, 1);
+	layout->addWidget(game_open, 0, 2);
+
+	m_dvd_edit = new QLineEdit(Settings().GetDVDRoot());
+	connect(m_dvd_edit, &QLineEdit::editingFinished,
+			[=]{ Settings().SetDVDRoot(m_dvd_edit->text()); });
+	QPushButton* dvd_open = new QPushButton;
+	connect(dvd_open, &QPushButton::clicked, this, &PathDialog::BrowseDVDRoot);
+	layout->addWidget(new QLabel(tr("DVD Root")), 1, 0);
+	layout->addWidget(m_dvd_edit, 1, 1);
+	layout->addWidget(dvd_open, 1, 2);
+
+	m_app_edit = new QLineEdit(Settings().GetApploader());
+	connect(m_app_edit, &QLineEdit::editingFinished,
+			[=]{ Settings().SetApploader(m_app_edit->text()); });
+	QPushButton* app_open = new QPushButton;
+	connect(app_open, &QPushButton::clicked, this, &PathDialog::BrowseApploader);
+	layout->addWidget(new QLabel(tr("Apploader")), 2, 0);
+	layout->addWidget(m_app_edit, 2, 1);
+	layout->addWidget(app_open, 2, 2);
+
+	m_nand_edit = new QLineEdit(Settings().GetWiiNAND());
+	connect(m_nand_edit, &QLineEdit::editingFinished,
+			[=]{ Settings().SetWiiNAND(m_nand_edit->text()); });
+	QPushButton* nand_open = new QPushButton;
+	connect(nand_open, &QPushButton::clicked, this, &PathDialog::BrowseWiiNAND);
+	layout->addWidget(new QLabel(tr("Wii NAND Root")), 3, 0);
+	layout->addWidget(m_nand_edit, 3, 1);
+	layout->addWidget(nand_open, 3, 2);
+
+	return layout;
+}
+
+QHBoxLayout* PathDialog::MakeOkButtonLayout()
+{
+	QHBoxLayout* hlayout = new QHBoxLayout;
+	hlayout->addStretch();
+	QPushButton* ok = new QPushButton(tr("Close"));
```

Comment:
In general, it's good practice to use QDialogButtonBox because different desktop environments use different orders of the "Ok", "cancel", etc buttons, and QDialogButtonBox automatically chooses the proper order to follow the HIG of the current target environment. If this is different to what DolphinWX is currently doing, then DWX just integrates incorrectly with the environment.

Other than that, there may be a few other subtle advantages like that mentioned in the Qt docs.